cognite/client/_api/iam.py

```diff
@@ -187,17 +187,49 @@ def __init__(self, config: ClientConfig, api_version: str | None, cognite_client
         super().__init__(config, api_version, cognite_client)
         self._LIST_LIMIT = 100
 
-    def create(self, client_credentials: ClientCredentials | None = None) -> CreatedSession:
+    def __call__(self, client_credentials: ClientCredentials | dict | None = None) -> CDFSession:
+        """
+        Create a session and return a context manager that will revoke the session when it exits.
+
+        Args:
+            client_credentials (ClientCredentials | dict | None): The client credentials to create the session. If set to None,
+                a session will be created using the credentials used to instantiate -this- CogniteClient object.
+                If that was done using a token, a session will be created using token exchange.
+                Similarly, if the credentials were client credentials, a session will be created using client credentials.
+                This method does not work when using client certificates (not supported server-side).
+
+        Returns:
+            CDFSession: The context manager that will revoke the session when it exits.
+
+        Examples:
+
+            Create a session and use it in a context manager:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> with c.iam.sessions() as session:
+                ...     print(session.status)
+        """
+
+        return CDFSession(self, client_credentials)
+
+    def create(self, client_credentials: ClientCredentials | dict | None = None) -> CreatedSession:
```

Comment:
Let's add a note about discouraging usage of this method since revoking is not handled. Should use the ctx manager instead unless you know what you're doing.

Reply:
Perhaps my understanding is wrong, but I don't see why I would use the auto-revoking context manager. If I create a session, it is because I want to pass the nonce to some service that binds it (whatever that means) and continues running on my behalf; like a transformation or a function schedule. Wouldn't the act of revoking effectively stop the service shortly after? 🤔

Reply:
Well, clearly I’m misunderstanding what revoking means; just had a function schedule complete successfully 8 hours after revoking the session it was created from… 🤔 

So what exactly does revoking imply I wonder?

Reply:
Right, so this is only the desired behaviour in integration tests then. I've probably misunderstood how sessions were being used - thought you would create one session per function run, but it's one per function?

Reply:
My understanding of it from workflows is that the session is used to trigger a workflow, but once it is triggered you don't need the session to be valid to continue running the workflow. I would guess it is similar for Functions.

Reply:
Aight, I've checked a bit more into this now (with some help from @ngerstle-cognite ❤️ ) and can confirm that revoking the session *will inevitably kill the access to CDF*. This can take up to 45 min / 1 hour. The reason my function kept running nicely was that I actually didn't make any CDF calls in my function... Yes, huge 🤦 on my part 😂 

So, changes we need to make to this PR:
- Remove the context manager from the public methods and move to test tooling.
- Then revert the changes to `FunctionSchedulesAPI.create`...
- ...and `FunctionsAPI.call`. For this one, we could keep the changes for when `wait` is passed as `True`, since this would not kill the session before the function was finished. However, I think a better solution is to not add this kind of complexity into the SDK.